src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -208,6 +210,13 @@ public void setBasedir(String basedir) {
         this.basedir = basedir;
     }
 
+    @Override
+    public int process(Collection<Path> files) throws CheckstyleException {
+        return process(files.stream()
+                .map(Path::toFile)
```

Comment:
As we driving to better/modern class usage, we should use Path as primary, and old method that use File so convert to File. Right?

Reply:
yes, we can do everything in dedicated request.

Reply:
please do in this PR.

Reply:
have done already its huge. Why no split? This is not good style.

Reply:
- https://github.com/checkstyle/checkstyle/pull/16978
- https://github.com/checkstyle/checkstyle/pull/16977
- https://github.com/checkstyle/checkstyle/pull/16968

need to follow patterns always not sometime. give dedication following SPOT, SOC, like seen in 

- https://github.com/apache/maven/pull/2326